pyre/extra_tests/parity_tests/with_exit_suppression_after_branch.py

```
@@ -0,0 +1,158 @@
+    """The failing shape: the second block's `__exit__` was skipped."""
+    log = []
+    cm = Suppress(log)
+    taken = 0
+    escaped = 0
+    for i in range(ROUNDS):
+        if i % 100 - 50 > 0:
+            taken += 1
+        else:
+            try:
+                with cm:
+                    raise ValueError(1)
```

Ruff (0.16.2)

[warning] Abstract `raise` to an inner function

(TRY301)